mmv1/api/product.go

```diff
@@ -68,6 +69,9 @@ type Product struct {
 	LegacyName string `yaml:"legacy_name,omitempty"`
 
 	ClientName string `yaml:"client_name,omitempty"`
+
+	// The compiler to generate the downstream files, for example "terraformgoogleconversion-codegen".
+	Compiler string `yaml:"-"`
```

Comment:
I was thinking it may be worthwhile to have a more universal way to check which MMv1 compiler/provider a template is being called from. But I guess all inputs into templates so far are either a resource which has a reference to the product object, or your new template that takes in the product itself.

Reply:
Yeah- I just shimmed this for now, but I was thinking we may want to introduce a GeneratorSettings/GeneratorRunSettings struct that we [embed](https://gobyexample.com/struct-embedding) in the api types & fill in during shared Generate code. Gets rid of the opportunity for drift/error.